prov/sockets/src/sock_conn.c

```diff
@@ -130,30 +130,32 @@ void sock_conn_map_destroy(struct sock_ep_attr *ep_attr)
 	int i;
 	struct sock_conn_map *cmap = &ep_attr->cmap;
 	for (i = 0; i < cmap->used; i++) {
-		sock_epoll_del(&cmap->epoll_set, cmap->table[i].sock_fd);
-		sock_pe_poll_del(ep_attr->domain->pe, cmap->table[i].sock_fd);
-		ofi_close_socket(cmap->table[i].sock_fd);
+		if (cmap->table[i].sock_fd != -1) {
+			sock_pe_poll_del(ep_attr->domain->pe, cmap->table[i].sock_fd);
+			sock_conn_release_entry(cmap, &cmap->table[i]);
+		}
 	}
 	free(cmap->table);
 	cmap->table = NULL;
 	cmap->used = cmap->size = 0;
 	sock_epoll_close(&cmap->epoll_set);
-	fastlock_destroy(&cmap->lock);
```

Comment:
It feels more natural to have the cmap lock destroyed in the function that destroy the cmap, just as it's initialized in the function that initialize the cmap.